pkg/awsclient/client.go

```diff
@@ -376,15 +407,22 @@ func newClient(controllerName, awsAccessID, awsAccessSecret, token, region strin
 	if controllerName != "" {
 		// NOTE(efried): We can't count on the returned client being used by a single thread, so we
 		// create a separate timer for each request.
-		timers := make(map[*request.Request]time.Time)
+		timers := &timerMap{
+			theMap: make(map[string]time.Time),
+		}
 		// Validate is the first phase of a request. Start the timer as early as possible (as the first handler there).
 		s.Handlers.Validate.PushFront(func(r *request.Request) {
-			timers[r] = time.Now()
+			timers.start(r.RequestID)
```

Comment:
Initial testing shows that this RequestID isn't set at this point. I am sad. I'll try using a later-stage Handler. If that doesn't work, I may go back to using the Request's memory address -- that may be okay if I just add the thread safety -- but I sure would like to find something better.